js/tabs.js

```diff
@@ -326,6 +309,54 @@ Tabs.prototype.openFiles = function() {
       this.openFileEntry.bind(this));
 };
 
+/**
+ * @param {function()} callback
+ * Invoke the save dialog for all tabs with unsaved progress. Does not close any tabs.
+ */
+Tabs.prototype.promptAllUnsaved = function(callback) {
+  this.promptAllUnsavedFromIndex_(0, callback);
+}
+
+Tabs.prototype.promptAllUnsavedFromIndex_ = function(i, callback) {
+  if (this.tabs_.length > i) {
```

Comment:
Nit: How about bailing out early?

```js
if (this.tabs_.length === 0) {
  callback();
  return;
}
...
```

Reply:
doesn't the current code already do that? if tabs_.length is 0, we immediately hit the else block below that calls callback()